internal/fleet/agent_policy/models.go

```diff
@@ -605,3 +862,162 @@ func mergeAgentFeature(existing []apiAgentFeature, newFeature *apiAgentFeature)
 
 	return &result
 }
+
+// convertHttpMonitoringEndpointToAPI converts the HTTP monitoring endpoint config to API format
+func (model *agentPolicyModel) convertHttpMonitoringEndpointToAPI(ctx context.Context) (*struct {
+	Buffer *struct {
```

Comment:
Should we pull these out to type aliases?

Reply:
https://github.com/elastic/terraform-provider-elasticstack/pull/1537/commits/cdb2794f54ae7bc616e2b238cbe5d049ed7bab08